torch/cuda/__init__.py

```diff
@@ -605,3 +610,9 @@ class ComplexFloatStorage(_CudaBase, torch._C.CudaComplexFloatStorageBase, _Stor
 from . import profiler
 from . import nvtx
 from . import amp
+
+from torch._C import register_device_runtime, set_default_device_type
+
+
+current_module = sys.modules[__name__]
+register_device_runtime('cuda', current_module)
```

Comment:
user can register new runtime in python, or in C++ native, right?

Reply:
Yes. User can register the new runtime in C++ thru `void register_device_runtime(const DeviceType dev_type, PyObject* module)`